eng/build.sh

```diff
@@ -169,9 +169,8 @@ while [[ $# > 0 ]]; do
     --warnaserror)
       warn_as_error=true
       ;;
-    --sourcebuild|/p:arcadebuildfromsource=true)
-      # Arcade specifies /p:ArcadeBuildFromSource=true instead of --sourceBuild, but that's not developer friendly so we
-      # have an alias.
+    --sourcebuild|/p:dotnetbuildsourceonly=true)
```

Comment:
Should we also handle `DotNetBuildRepo=true`?

Reply:
I think it's unlikely that your team will pass that separately. Also, that passing that alone would imply a non-source-only VMR style build. Do you know whether anyone uses that alias?

Reply:
I mean, I think we don't use either alias, it's just the arcade template that does :laughing:.

Reply:
Alright I'll investigate further. I didn't think that the VMR actually invokes this script.

Reply:
https://github.com/dotnet/roslyn/blob/09ef0c03d033b17f0e5750f265275ed6d1f3cca7/eng/common/core-templates/steps/source-build.yml#L89-L91 was the only place I found that used them.

Reply:
Alright so the crux of why this was necessary is that static graph restore gets disabled under that /p: alias. I moved this to the inner source build arg creation.